Unity-Technologies/com.unity.multiplayer.samples.coop
# feat: other players loading progress in loading screen [MTT-2239]

### Description (*)

This PR modifies the loading screen so that it can also display the loading progress of other players. To accomplish this, I added a simple player-owned NetworkObject that contains an owner-writtable NetworkVariable to hold the player's current progress. These are spawned by LoadingProgressManager on the server and kept in a map on all clients, so that a client can update its progress and read the progress of the other clients.
### Related Pull Requests

### Issue Number(s) (*)

Fixes issue(s): [MTT-2239](https://jira.unity3d.com/browse/MTT-2239)
### Manual testing scenarios

1. Start a game as a host
2. Connect clients to the session
3. Complete character selection and move to the next scene
4. Disconnect a client
5. Connect a new client
6. Go to Post-game
7. Leave the game session
The loading screen should be properly displayed all those times, and should display the progress of other connected players. The progress bars should update when characters disconnect and when new ones connect.
### Questions or comments

### Contribution checklist
